app/stac_api/models.py

```diff
@@ -197,27 +190,469 @@ def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asse
+                        ).earliest('properties_start_datetime').properties_start_datetime
+                    else:
+                        earliest_start_datetime = None
+                    # get earliest datetime, or none in case none exists.
+                    # This should be the case, if the item that is currently
+                    # being updated is the only item in the collection or
+                    # no item has properties.datetime defined but all do have
+                    # only ranged (start_ and end_datetimes defined)
+                    if bool(qs.filter(properties_datetime__isnull=False)):
+                        earliest_datetime = qs.filter(
+                            properties_datetime__isnull=False
+                        ).earliest('properties_datetime').properties_datetime
+                    else:
+                        earliest_datetime = None
+
+                    if earliest_start_datetime is None and earliest_datetime is None:
+                        # currently updated item is the only one:
+                        self.cache_start_datetime = new_start_datetime
+                    elif earliest_start_datetime is not None:
+                        self.cache_start_datetime = earliest_start_datetime
+                    else:
+                        self.cache_start_datetime = earliest_datetime
+            elif new_start_datetime < self.cache_start_datetime:
+                # item's start_datetime did not define the left bound of the
+                # collection's temporal extent before update, which does not
+                # matter anyways, as it defines the new left bound after update
+                # and collection's start_datetime can be simply adjusted
+                self.cache_start_datetime = new_start_datetime
+
+        self.save()
+
+    def update_item_end_datetime(
+        self,
+        old_start_datetime,
+        new_start_datetime,
+        old_end_datetime,
+        new_end_datetime,
+        item_id,
+        qs=None
+    ):
+        '''
+        This function is called from within update_temporal_extent() when an
+        item in the collection is updated to check if the collection's
+        end_datetime needs to be updated.
+        :param old_start_datetime: item's old value for properties_start_datetime
+        :param new_start_datetime: item's updated value for properties_start_datetime
+        :param old_end_datetime: item's old value for properties_end_datetime
+        :param new_end_datetime: item's updated value for properties_end_datetime
+        :param item_id: the id of the item being treated
+        :param qs: (optional) queryset with all items of the collection excluding the one
+        being updated.
+        '''
+        logger.debug(
+            'Updating item %s in collection (adjusting end_datetime) %s',
+            item_id,
+            self.collection_name
+        )
+        if old_end_datetime != new_end_datetime:
```

Comment:
Noticed that this can be removed, as it is redundant. This function is only called, when this condition is true (see line 578)

Reply:
removed redundant if statement